services/github/github-issue-detail.service.js

```diff
@@ -1,27 +1,137 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const {
-  makeLabel: getLabel,
-  makeBadgeData: getBadgeData,
-} = require('../../lib/badge-data')
-const { makeLogo: getLogo } = require('../../lib/logos')
+const Joi = require('joi')
+const { InvalidResponse } = require('..')
+const { nonNegativeInteger } = require('../validators')
 const { formatDate } = require('../text-formatters')
-const { age: ageColor } = require('../color-formatters')
+const { age } = require('../color-formatters')
+const { GithubAuthService } = require('./github-auth-service')
 const {
   documentation,
-  stateColor: githubStateColor,
-  commentsColor: githubCommentsColor,
-  checkErrorResponse: githubCheckErrorResponse,
+  errorMessagesFor,
+  stateColor,
+  commentsColor,
 } = require('./github-helpers')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class GithubIssueDetail extends LegacyService {
+const getDefaultLabel = ({ json }) => {
```

Comment:
I decided to go with a solution of leveraging objects for the various `which` types to enable usage of more granular/stricter schemas (vs a single schema with all properties as optional) and to avoid needing multiple switch statements.

Reply:
This is a bit difficult to follow, though it doesn't look too hard to maintain – and this code is doing a lot of stuff.

One suggestion that I think would make this easier to understand is to implement render functions inside your objects to make them more direct, instead of breaking up the render responsibility into separate functions that get called later. There might be the slightest bit more repetition with render functions, though I think it will be quite a bit easier to follow.

Reply:
> to implement render functions inside your objects to make them more direct

I like the sound of that, will do.

The other thing i considered was moving all of this `which` specific transform/render into a separate file, but decided to keep it all together for now. Any thoughts/preferences on that (in-file w/service class vs. separate file)?

Reply:
Probably in with the service.

Reply:
Done! I like the way this looks a lot better, it was a really good idea. I believe it's far more readable now with each map object containing its respective `schema`, and functions for `render` and `transform`